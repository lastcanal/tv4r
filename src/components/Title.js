import React from 'react'
import { makeStyles } from '@material-ui/core/styles'
import PropTypes from 'prop-types'

const useStyles = makeStyles(({ spacing, palette }) => ({
  root: {
    padding: spacing(2),
  },
  title: {
    fontSize: 22,
    textDecoration: 'none',
    cursor: 'pointer',
    color: palette.text.primary,
    textOverflow: 'ellipsis',
    overflow: 'hidden',
    whiteSpace: 'nowrap',
  },
}))

const Title = ({ post }) => {
  const classes = useStyles()
  if (!post) return ''

  return (
    <div className={classes.root}>
      <a
        className={classes.title}
        target="_BLANK"
        rel="noopener noreferrer"
        href={'https://reddit.com' + post.permalink}
      >
        {post.title ? post.title : ''}
      </a>
    </div>
  )
}

Title.propTypes = {
  post: PropTypes.object,
}

export default Title
