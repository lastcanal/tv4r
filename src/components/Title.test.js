import Title from './Title'

describe('Title', () => {

  it('should render Title', () => {
    const enzymeWrapper = mount(
      <Title post={{title: "hello"}} />
    );

    expect(enzymeWrapper).toMatchSnapshot()
  })

  it('should render without title', () => {
    const enzymeWrapper = mount(
      <Title post={{title: undefined}} />
    );

    expect(enzymeWrapper).toMatchSnapshot()
  })

  it('should render without post', () => {
    const enzymeWrapper = mount(
      <Title post={undefined} />
    );

    expect(enzymeWrapper).toMatchSnapshot()
  })

})
