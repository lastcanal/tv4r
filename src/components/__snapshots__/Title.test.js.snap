// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title should render Title 1`] = `ReactWrapper {}`;

exports[`Title should render without post 1`] = `ReactWrapper {}`;

exports[`Title should render without title 1`] = `ReactWrapper {}`;
